sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/MetaDataDiagnosticsContext.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.cosmos.ZonedDateTimeSerializer;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MetaDataDiagnosticsContext {
+    public volatile List<MetaDataDiagnostic> metaDataDiagnosticList;
+
+    public void addMetaDataDiagnostic(MetaDataDiagnostic metaDataDiagnostic) {
+        if (metaDataDiagnosticList == null) {
+            metaDataDiagnosticList = new ArrayList<>();
+        }
+
+        metaDataDiagnosticList.add(metaDataDiagnostic);
```

Comment:
I am guessing this can be accessed from multiple threads. Then shouldn't this be a synchronized list?

see this similar race condition bug in the diagnostics string: https://github.com/Azure/azure-sdk-for-java/pull/8505